request-manager/grapher/grapher.go

```diff
@@ -410,9 +410,16 @@ func (o *Grapher) buildComponent(name string, nodeDefs map[string]*NodeSpec, nod
 		if nodesCount == len(nodesToBeDone) {
 			missingArgsNodes := []string{}
 			for n, _ := range nodesToBeDone {
-				missingArgsNodes = append(missingArgsNodes, n.Name)
+				// Figure out which args are missing
+				missingArgs := []string{}
+				for _, arg := range n.Args {
+					if _, ok := nodeArgs[arg.Expected]; !ok {
```

Comment:
You should be checking arg.Given, not arg.Expected. Also, looking at [allArgsPresent](https://github.com/square/spincycle/blob/c3b603cc7bbcf5a6552f1bc0ba345a2091b2e66c/request-manager/grapher/grapher.go#L582), I think you also need to look at `each` and `if` for missing args.